### ISSUE TITLE
Automate validation of recipe notebooks

### ISSUE THREAD

**Issue by adampingel** [ibm-granite-community/pm#41]

**Comment by rawkintrevo** [comment]
as far as i can tell _none_ of the notebooks currently work in colab. and if they don't work in colab, they won't work for testing. 

I think with a fair bit of surgery, I _could_ get the getting started one working in colab (and therefor testable). 

The shell/ollama based ones require significant resources, that will blow out free tier Colab, and GH Action limits
The 'jupyter working locally' was never meant to run in colab, so I'll give that one a miss
the vscode thing, also not really designed for colab. 
*NOTE: i don't use vscode or jupyter local, so I can't really test those either. 

tl;dr
At a minimum I need to pull those open in colab buttons off of everything. That's jsut setting people up for failure. 
In theory, fix getting started to work in colab, then set it up for testing. The test _will_ be flakey though (as replicate is sometimes flaky)

**Comment by fayvor** [comment]
@rawkintrevo that makes sense. It makes me think we should separate the recipes for running models locally from those that use remote models, and only link to colab for the remote recipes. 

And we should probably default to remote recipes, making a duplicate recipe for running the model locally, if appropriate.

**Comment by rawkintrevo** [comment]
while working on #116 i believe i stumbled on a way to make ollama work in colab (and so possibly git actions), i will pursue today

**Comment by rawkintrevo** [comment]
implemented here. https://github.com/ibm-granite-community/utils/blob/main/examples/Basic%20Usage.ipynb
working on testing now

**Comment by rawkintrevo** [comment]
Ollama testing POC implemented in `utils` repo. 

qid: https://github.com/ibm-granite-community/utils/actions/runs/10744651782/job/29802287111

Next step will be updating existing recipes / updating cookbook repo

**Comment by rawkintrevo** [comment]
https://github.com/ibm-granite-community/granite-code-cookbook/pull/75
